### DISCUSSION TITLE
Mathematics extension

### DISCUSSION THREAD

**Discussion by caph1993** [remirror/remirror#810] [Ideas]
Dear developers.

There is a mathematics extension for prosemirror called [prosemirror-math](https://github.com/benrbray/prosemirror-math) that uses [katex](https://katex.org/). I think this could be a nice extension for remirror as well :+1: 

It should not very complicated to implement because prosemirror-math is already written in Typescript, is short, and is very readable. However, I have not succeeded by myself because I am still very new to this.

Formulas and variables are crucial for some people, although definitely not for everybody. I think it would be great to have such extension in remirror. Don't you agree? It is very simple and it looks great:

![Inline mode][inline]

![Display mode][display]

[inline]: https://github.com/benrbray/prosemirror-math/raw/master/img/prosemirror-math_inline.gif "Inline mode"
[display]: https://github.com/benrbray/prosemirror-math/raw/master/img/prosemirror-math_insert-display.gif "Display mode"

**Comment by xero0001** [comment]
I would love to see this.

**Comment by ifiokjr** [comment]
This looks like a great addition. I'd be happy to support anyone who's interested in adding the extension.

**Comment by vighnesh153** [comment]
I want this too.

**Comment by jpribyl** [comment]
Hmm, I wrote this a while back.. https://github.com/jpribyl/react-hook-mathjax

Maybe I'll take a stab at porting it. I think mathjax has slightly better coverage and accessibility than katex.
Probably it makes sense to split it apart by "inline" and "block" similar to the code extensions.

One thing that would be a good deal more work but _really_ cool is if there was support for "math buttons" which prefill some bits and let you write out equations without needing to know latex.

Not really sure how to implement that.... anybody done something like that before?


EDIT I think this is totally possible, it would essentially be a form that prefills some of the latex for you. Could still get tricky though
![image](https://user-images.githubusercontent.com/27659352/152469518-e7b6ca17-f03e-448a-8dd3-a6fda8b760c3.png)

**Comment by Aidenzich** [comment]
Hey, I was wondering if there is a solution here to support KaTeX input?
